### PR TITLE
add a flag to control whether delete data of dropped space when restart

### PR DESCRIPTION
**Pull request by critical27**
1. Add a flag to control whether delete data of dropped space when restart
2. Identify a potential bug in raft, we don't modify in 1.0 for safety

### CODE REVIEW

**Review by liuyu85cn — DISMISSED** [review]
LGTM

File: src/kvstore/NebulaStore.cpp [hunk 1]
```diff
@@ -72,14 +73,16 @@ bool NebulaStore::init() {
...
-                                  << " does not exist any more, remove the data!";
-                        auto dataPath = folly::stringPrintf("%s/%s",
-                                                            rootPath.c_str(),
-                                                            dir.c_str());
-                        CHECK(fs::FileUtils::remove(dataPath.c_str(), true));
+                        if (FLAGS_auto_remove_invalid_space) {
+                            // TODO We might want to have a second thought here.
+                            // Removing the data directly feels a little strong
+                            LOG(INFO) << "Space " << spaceId
+                                    << " does not exist any more, remove the data!";
```
Comment (darionyaphet) [inline src/kvstore/NebulaStore.cpp:80]: alignment ?

Comment (critical27) [reply src/kvstore/NebulaStore.cpp]: fixed

**Review by darionyaphet — APPROVED** [review]

**Review by panda-sheep — APPROVED** [review]
LGTM

**Review by darionyaphet — APPROVED** [review]

**Pull request merged**